gpm_core/src/package.rs

```diff
@@ -57,6 +81,48 @@ impl PackageInformation {
             extra_data: Vec::new(),
         }
     }
+
+    /// return all the information required for published package
+    /// ([`RequiredPublishInformation`]), otherwise, return an error containing a
+    /// list of the missing field
+    pub fn get_required_publish_information(
```

Comment:
Doing the checks manually is simple and clear enough, but I wonder would it be possible to have an attribute of some sort on the `StoredPackageInformation` fields that indicates a field is required for publishing, but optional otherwise?

That would allow us to define the package fields in a single place, without risking that a developer forgets to edit this section here accordingly.

Alternatively if you can make rust compiler ensure that this function exhausts all the possibilities in the `StoredPackageInformation`, that would work too

Reply:
AFAIK, it will need some macro, and this is probably too much work for this. I will however add a comment to help on renaming/adding/removing required entry (only the case of renaming and adding data can lead to unexepected value, removing one would cause a compilation error).

Reply:
Alright, good to know 👍 This isn't a blocker issue so feel free to merge, but maybe @Veykril can provide insight on how something like this could be achieved just for future reference.